src/platform/src/Transport/Dsn.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\AI\Platform\Transport;
+
+use Symfony\AI\Platform\Exception\InvalidArgumentException;
+
+final class Dsn
```

Comment:
Where does this code com from? We have several `DSN` classes in `symfony/symfony` which should be used. I propose to copy it from `symonfy/notifer` including tests.